rust-crates/symblib-capi/src/gosym.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{FfiResult, StatusCode, SymblibPointResolver};
+use std::ffi::{c_char, CStr};
+use std::path::Path;
+use symblib::symbconv::PointResolver;
+use symblib::{gosym::GoRuntimeInfo, objfile};
+
+pub struct SymblibGoRuntime {
+    #[allow(dead_code)]
+    runtime: GoRuntimeInfo<'static>,
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn symblib_goruntime_new(
+    executable: *const c_char,
+    runtime: *mut *mut SymblibPointResolver,
+) -> StatusCode {
+    match goruntime_new_impl(executable, runtime) {
+        Ok(()) => StatusCode::Ok,
+        Err(e) => e,
+    }
+}
+
+unsafe fn goruntime_new_impl(
+    executable: *const c_char,
+    runtime: *mut *mut SymblibPointResolver,
+) -> FfiResult {
+    let executable = CStr::from_ptr(executable)
+        .to_str()
+        .map(Path::new)
+        .map_err(|_| StatusCode::BadUtf8)?;
+
+    let obj = Box::leak(Box::new(objfile::File::load(executable)?));
```

Comment:
This leaks memory. Hmmmm. You may have to do something ugly like:

```
pub struct SymblibGoRuntime {
    objfile: Box<objfile::File>,
    runtime: GoRuntimeInfo<'static>,
}

impl PointResolver for SymblibGoRuntime {
   fn symbols_for_pc(&self, pc: VirtAddr) -> Result<Vec<ResolvedSymbol>> {
      self.runtime.symbols_for_pc(pc)
   }
}
```

Essentially: create a wrapper that owns and frees the backing object file on drop and then return that in the `SymblibPointResolver`.